custom/onse/tasks.py

```diff
@@ -83,9 +87,54 @@ def update_facility_cases_from_dhis2_data_elements(
         otherwise they are emailed.
 
     """
+    _update_facility_cases_from_dhis2_data_elements(period, print_notifications)
+
+
+@task
+def _update_facility_cases_from_dhis2_data_elements(period, print_notifications, retry_attempt=0):
     if not domain_exists(DOMAIN):
         return
     dhis2_server = get_dhis2_server(print_notifications)
+
+    if _server_is_online(dhis2_server):
+        _execute_update_facility_cases_from_dhis2_data_elements(dhis2_server, period, print_notifications)
+    else:
+        retry_attempt += 1
+        if retry_attempt <= MAX_RETRY_ATTEMPTS:
+            retry_in_n_days = 2 ** (retry_attempt - 1)
+            retry_date = datetime.utcnow() + relativedelta(days=retry_in_n_days)
+
+            print(f'Importing ONSE ISS facility cases from DHIS2 failed: server unreachable. '
+                  f'Retrying on {retry_date.date()}')
+
+            schedule_execution(
+                _update_facility_cases_from_dhis2_data_elements,
+                [period, print_notifications, retry_attempt],
+                retry_date
+            )
+        else:
+            print('Importing ONSE ISS facility cases from DHIS2 failed: server unreachable. '
+                  'No more retry attempts')
+
+
+def schedule_execution(on_date: datetime, args: list, callback_task):
+    always_eager = hasattr(settings, "CELERY_TASK_ALWAYS_EAGER") and settings.CELERY_TASK_ALWAYS_EAGER
+    settings.CELERY_TASK_ALWAYS_EAGER = False
+    callback_task.apply_async(tuple(args), eta=on_date)
+    settings.CELERY_TASK_ALWAYS_EAGER = always_eager
+
+
+def _server_is_online(dhis2_server: ConnectionSettings) -> bool:
+    network_locality = urlparse(dhis2_server.url).netloc
+    domain = network_locality.split(':')[0]
+    return os.system('ping -c 1 ' + domain) == PING_SUCCESS_CODE
```

Comment:
Is sending one packet reliable enough?

Reply:
:shrug: Best of three? I'm not sure whether you get a success code on one successful ping, or whether you need all three to be successful.

Reply:
... Alternatively, you could avoid the system call by using a socket:

```python
    try:
        sock = socket.create_connection((domain, 80), timeout=5)
    except:
        return False
    sock.shutdown()
    sock.close()
    return True
```

Reply:
@snopoke thanks for the reference!